tsdb/engine/tsm1/iterator.gen.go.tmpl

```diff
@@ -275,6 +277,10 @@ func (itr *{{.name}}Iterator) Next() (*query.{{.Name}}Point, error) {
 			itr.m[itr.conds.names[i]] = itr.conds.curs[i].nextAt(seek)
 		}
 
+		if itr.opt.Condition != nil {
```

Comment:
Can you describe what's going on in a comment?  This conditional seems very wide; are there `itr.opt.Condition` that require a time field and ones that don't?  This question may be a product of my lack of understanding.

Reply:
I'll add an description but for for the most part this is similar to: https://github.com/influxdata/influxdb/pull/27005/files#diff-3f76e91fed19b415e937d7ea1a37cf166e3d01da63db8aef20a0addf6948aa71R206-R208

I want to ensure there is a `Condition` so I'm not setting the map every time we call in to this code 

The Condition has `date_part` nested so I could check specifically for `date_part` with a bit of extra code that may be more verbose: 
<img width="466" height="348" alt="Screenshot 2025-12-10 at 3 45 36 PM" src="https://github.com/user-attachments/assets/cee16ddb-9710-4951-9819-3d5b75cd534f" />

Reply:
the date_part can be on either side of the condition, right?  So it would be verbose to find it, as you mention.  What's the cost of an unnecessary time set?  Cheap or costly?  Memory cost, if any?

Reply:
@davidby-influx It won't be very costly as this would just cause an 8 byte allocation whenever it gets called. The operation for inserting a map value in Go should also be incredibly performant. But I guess if we have millions of queries being ran there is a possibility of a bit more memory usage, unless the GC takes care of it fast enough -- but that would require higher CPU utilization.